File: ABNLookupSampleCodeVBDotNet/ABRXMLSearchClient/ProxyXMLRPCSearch.vb
Imports ABRXMLSearchClient.WebReferenceRPC
'Imports ABRXMLSearchClient.au.gov.business.abr

' Strongly Typed Search - Uses the web service proxy classes to search ABRpublic. 
Public Class ProxyXMLRPCSearch
   '-----------------------------------------------------------------------------------------------
   ' Search by ABN
   '-----------------------------------------------------------------------------------------------
    Public Function ABNSearch(ByVal searchString As String, _
                              ByVal history As String, _
                              ByVal guid As String) As WebReferenceRPC.Payload

        Dim ABRSearch As WebReferenceRPC.ABRXMLSearchRPC = New WebReferenceRPC.ABRXMLSearchRPC
      Return ABRSearch.ABRSearchByABN(searchString, history, guid)
    End Function
   '-----------------------------------------------------------------------------------------------
   ' Search by ASIC
   '-----------------------------------------------------------------------------------------------
    Public Function ASICSearch(ByVal searchString As String, _
                              ByVal history As String, _
                              ByVal guid As String) As WebReferenceRPC.Payload

        Dim ABRSearch As WebReferenceRPC.ABRXMLSearchRPC = New WebReferenceRPC.ABRXMLSearchRPC
        Return ABRSearch.ABRSearchByASIC(searchString, history, guid)

    End Function
   '-----------------------------------------------------------------------------------------------
   ' Search by Name
   '-----------------------------------------------------------------------------------------------
    Public Function NameSearch(ByVal guid As String, _
                               ByVal criteria As WebReferenceRPC.ExternalRequestNameSearch) As WebReferenceRPC.Payload

        Dim ABRSearch As WebReferenceRPC.ABRXMLSearchRPC = New WebReferenceRPC.ABRXMLSearchRPC
        Return ABRSearch.ABRSearchByName(criteria, guid)
    End Function

End Class

File: ABNLookupSampleCodeCSharp35/SampleAbnLookupClient/ProxyXmlRpcSearch.cs
using System;
using AbnLookup.SearchClientCSharpe.ServiceReferenceAbnLookupRpc;

namespace AbnLookup.SearchClientCSharpe {
   class ProxyXmlRpcSearch {
      // -----------------------------------------------------------------------------------------------
      // Private constructor
      // -----------------------------------------------------------------------------------------------
      private ProxyXmlRpcSearch() { }
      // -----------------------------------------------------------------------------------------------
      //  Search by ABN
      // -----------------------------------------------------------------------------------------------
      public static Payload AbnSearch(string searchText, string history, string guid) {
         ABRXMLSearchRPCSoapClient Search = new ABRXMLSearchRPCSoapClient();
         return Search.ABRSearchByABN(searchText, history, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by ASIC
      // -----------------------------------------------------------------------------------------------
      public static Payload AsicSearch(string searchText, string history, string guid) {
         ABRXMLSearchRPCSoapClient Search = new ABRXMLSearchRPCSoapClient();
         return Search.ABRSearchByASIC(searchText, history, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by Name
      // -----------------------------------------------------------------------------------------------
      public static Payload NameSearch(string guid, ExternalRequestNameSearch criteria) {
         ABRXMLSearchRPCSoapClient Search = new ABRXMLSearchRPCSoapClient();
         return Search.ABRSearchByName(criteria, guid);
      }
      // -----------------------------------------------------------------------------------------------
      //  Search by Postcode
      // -----------------------------------------------------------------------------------------------
      public static Payload PostcodeSearch(string guid, string postcode) {
         ABRXMLSearchRPCSoapClient Search = new ABRXMLSearchRPCSoapClient();
         return Search.SearchByPostcode(postcode, guid);
      }
   }
}
